packages/core/tarball/src/getTarballDetails.ts

```diff
@@ -24,11 +24,15 @@ export async function getTarballDetails(buffer: Buffer): Promise<TarballDetails>
         next();
       })
       .on('finish', () => {
+        readable.destroy();
```

Comment:
not sure about this, in theory the stream Is closed by the client already because is piped, could you share more context? have you noticed an incorrect behaviour?

Reply:
Yes, it's closed automatically when done. However, I got some node warnings in the log related to garbage collection. Co-pilot suggested this as good practice.

Reply:
I'm still not convinced to destroy the readable stream on finish. I also think might make sense convert this to use `pipeline` instead to need to destroy ourselves.

Reply:
np. i will revisit if it becomes a bigger problem.